regolith/tools.py

```diff
@@ -1282,9 +1245,82 @@ def fragment_retrieval(coll, fields, fragment, case_sensitive = False):
                     break
     return ret_list
 
+
 def get_id_from_name(coll, name):
     person = fuzzy_retrieval(coll,["name", "aka", "_id"], name,
                              case_sensitive=False)
     if person:
         return person["_id"]
-    else: return None
\ No newline at end of file
+    else: return None
+
+
+def is_fully_appointed(person, begin_date=None, end_date=None):
+    """Checks if a collection of appointments for a person is valid and fully loaded
+        for a given interval of time
+
+        Parameters
+        ----------
+        person: dict
+            The person whose appointments need to be checked
+        begin_date: datetime, string, optional
+            The start date of the interval of time to check appointments for
+        end_date: datetime, string, optional
+            The end date of the interval of time to check appointments for
+
+        Returns
+        -------
+        bool:
+            True if the person is fully appointed and False if not
+
+        Examples
+        --------
+        >>> appts = [{"begin_year": 2017, "begin_month": 6, "begin_day": 1, "end_year": 2017, "end_month": 6,\
+         "end_day": 15, "grant": "grant1", "loading": 1.0, "type": "pd", }, {"begin_year": 2017, "begin_month": 6, \
+         "begin_day": 17,  "end_year": 2017,  "end_month": 6, "end_day": 30, "grant": "grant2", "loading": 1.0, \
+         "type": "pd",} ]
+        >>> aejaz = {"name": "Adiba Ejaz", "_id": "aejaz", "appointments": appts}
+        >>> is_fully_appointed(aejaz, "2017-06-01", "2017-06-30")
+
+        In this case, we have an insufficient loading and minimum of 0.0 on 2017-06-16, hence it would return False and
+        print "aejaz is short of full loading with minimim 0.0 on 2017-06-16".
+        """
+
+    status = True
+    appts = person.get('appointments')
+    earliest = get_dates(appts[0])['begin_date']
+    latest = get_dates(appts[0])['end_date']
+    for appt in appts:
+        dates = get_dates(appt)
+        if dates['begin_date'] < earliest:
+            earliest = dates['begin_date']
+        if dates['end_date'] > latest:
+            latest = dates['end_date']
+    begin_date = earliest if not begin_date else begin_date
+    end_date = latest if not end_date else end_date
+    if begin_date > end_date:
+        raise ValueError("invalid begin and end dates")
+    if isinstance(begin_date, str):
+        begin_date = date_parser.parse(begin_date).date()
+    if isinstance(end_date, str):
+        end_date = date_parser.parse(end_date).date()
+    timespan = end_date - begin_date
+    loading, max, min, help = [], 0.0, 1.0, ''
+    for x in range(0, timespan.days):
```

Comment:
fyi, you don't need the 0 here.  OK to have it, but personally I think it is more readable without tbh....

Reply:
done